### PR TITLE
adding guest to waiting room till host starts the board

### LINKED ISSUE #3: waiting room
only host can start the session, currently it starts as soon as connection is established

### PR DESCRIPTION
**Pull request by theVirtualMan**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -2,42 +2,9 @@
...
-    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
-    <!--
-      Notice the use of %PUBLIC_URL% in the tags above.
-      It will be replaced with the URL of the `public` folder during the build.
-      Only files inside the `public` folder can be referenced from the HTML.
-
-      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
```
Comment (Rajesh-Khanna) [inline public/index.html:26]: why is this removed?

File: src/apps/sketch/communication/HG.js [hunk 2]
```diff
@@ -152,8 +152,13 @@ export class Guest {
...
         // channel.onmessage = message => console.log({ channel: message.currentTarget.label, message: message.data });
-        if (Object.keys(this.channels).length >= 4) {
-            this.onConnection()
+        console.log("asf", channel.label)
+        if (channel.label === 'meta') {
+            channel.onmessage = (message) => {
+                if (message.data === 'active_board'){
+                    this.onConnection();
+                }
+            }
```
Comment (Rajesh-Khanna) [inline src/apps/sketch/communication/HG.js:161]: do not use on message channel only for one message, every thing from score to board activity like roles should be handled by meta channel

File: src/apps/sketch/communication/HG.js [hunk 3]
```diff
@@ -152,8 +152,13 @@ export class Guest {
         console.log({ channel });
         this.channels[channel.label] = channel;
         // channel.onmessage = message => console.log({ channel: message.currentTarget.label, message: message.data });
-        if (Object.keys(this.channels).length >= 4) {
-            this.onConnection()
+        console.log("asf", channel.label)
```
Comment (Rajesh-Khanna) [inline src/apps/sketch/communication/HG.js:155]: remove or rename this comment

File: src/apps/sketch/communication/HG.js [hunk 4]
```diff
@@ -152,8 +152,13 @@ export class Guest {
         console.log({ channel });
         this.channels[channel.label] = channel;
         // channel.onmessage = message => console.log({ channel: message.currentTarget.label, message: message.data });
-        if (Object.keys(this.channels).length >= 4) {
-            this.onConnection()
```
Comment (Rajesh-Khanna) [inline src/apps/sketch/communication/HG.js:156]: I added this so that I can confirm that all channels are created before starting the session, so that there wont be any channel undefined errors later

File: src/apps/sketch/index.jsx [hunk 5]
```diff
@@ -46,6 +46,7 @@ const Sketch = () => {
     }
 
     const startBoard = () => {
+        dataChannel.current.getChannel('meta').send('active_board');
```
Comment (Rajesh-Khanna) [inline src/apps/sketch/index.jsx:49]: store channel in useRef and use it to manage all coms

**Review by Rajesh-Khanna — CHANGES_REQUESTED** [review]

File: public/index.html (hunk 1 above)
Comment (theVirtualMan) [reply public/index.html:26]: Is it required?

Comment (theVirtualMan) [reply public/index.html:26]: we can add the exact fields while working on designing(css)

File: src/apps/sketch/communication/HG.js (hunk 2 above)
Comment (theVirtualMan) [reply src/apps/sketch/communication/HG.js:161]: This is for temporary purpose only. I will remove the event listener after getting the message. But when i am working on roles and score, i will move this part there

File: src/apps/sketch/communication/HG.js (hunk 3 above)
Comment (theVirtualMan) [reply src/apps/sketch/communication/HG.js:155]: sure

File: src/apps/sketch/communication/HG.js (hunk 4 above)
Comment (theVirtualMan) [reply src/apps/sketch/communication/HG.js:156]: don't worry I am going to use this line at some other place, we won't be able to start session if all the data channels aren't available

File: public/index.html (hunk 1 above)
Comment (Rajesh-Khanna) [reply public/index.html:26]: do not remove this maybe we can modify this text later but leave it as it is for now

File: src/apps/sketch/index.jsx (hunk 5 above)
Comment (theVirtualMan) [reply src/apps/sketch/index.jsx:49]: sure

File: src/apps/sketch/communication/HG.js (hunk 2 above)
Comment (Rajesh-Khanna) [reply src/apps/sketch/communication/HG.js:161]: no but we will need meta channel for showing all the participants in the room
